packages/relay/src/lib/types/spendingPlanConfig.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { SubscriptionTier } from '../db/types/hbarLimiter/subscriptionTier';
+
+/**
+ * Represents the configuration for a spending plan.
+ * Pre-configured spending plans are specified in a `spendingPlansConfig.json` file in the root of the project.
+ * @interface SpendingPlanConfig
+ */
+export interface SpendingPlanConfig {
+  /**
+   * The unique identifier for the spending plan.
+   */
+  id: string;
+
+  /**
+   * The name of the spending plan.
+   * @type {string}
+   */
+  name: string;
+
+  /**
+   * ETH addresses associated with the spending plan.
+   * @type {string[]}
+   * @optional
+   */
+  ethAddresses?: string[];
+
+  /**
+   * IP addresses associated with the spending plan.
+   * @type {string[]}
+   * @optional
+   */
+  ipAddresses?: string[];
+
+  /**
+   * The subscription tier associated with the spending plan.
+   * @type {SubscriptionTier}
+   */
+  subscriptionTier: SubscriptionTier;
+}
+
+/**
+ * Determines if the provided object is a valid spending plan configuration.
+ * @param plan - The object to validate.
+ * @returns {boolean} - True if the object is a valid {@link SpendingPlanConfig}, false otherwise.
+ */
+export function isValidSpendingPlanConfig(plan: any): plan is SpendingPlanConfig {
+  return (
+    plan &&
+    typeof plan.id === 'string' &&
```

Comment:
Should we validate `plan.id` against a UUID regex instead of just checking for type `string` to enhance security?

On another note, I still feel that requiring maintainers to manually construct a UUID is not the most convenient approach. For the initial loading of the configuration plan, maintainers would need to use an external tool to generate a UUID for each plan, and then continuously manage the UUID throughout the process, correct?

Is it possible to use the `name` field to look up spending plans instead? In real scenarios, each project or partner would typically have a unique name, and it's much easier to input and maintain the name of a project or organization than a randomly generated UUID. 

What are your thoughts on this?

Reply:
> Should we validate plan.id against a UUID regex instead of just checking for type string to enhance security?

No, as we discussed in the last meeting any unique string works and the relay operator will be responsible for what he selects for the ID, we can leave a recommendation for using a UUID string but it's up to them what they set it to.

> On another note, I still feel that requiring maintainers to manually construct a UUID is not the most convenient approach. For the initial loading of the configuration plan, maintainers would need to use an external tool to generate a UUID for each plan, and then continuously manage the UUID throughout the process, correct?

What do you mean by "continuously manage the UUID throughout the process" they only have to generate it once and enter it in the JSON file - there is no maintaining after that, it remains both in the JSON file and in the cache once it is created.

> Is it possible to use the name field to look up spending plans instead? In real scenarios, each project or partner would typically have a unique name, and it's much easier to input and maintain the name of a project or organization than a randomly generated UUID.

Using names as keys isn't ideal because they aren't guaranteed to be unique or stable—they can change or have duplicates, leading to cache issues. UUIDs are specifically designed for this purpose, ensuring reliable and consistent key generation for caching. 

Also, `"name"` is only an accessory field which is used only in the logs of HbarSpendingPlanConfigService` for better readability by people. These names are not saved in the cache or used anywhere else.

Reply:
> we can leave a recommendation for using a UUID string but it's up to them what they set it to
Ah, I see! I initially thought it had to be in UUID format, which is why I was concerned about the lack of constraints around it being a UUID.

All the other points make perfect sense, thanks for clarifying!